enhancements/sig-architecture/58-addon-configuration/README.md

```diff
@@ -0,0 +1,606 @@
+    // +optional
+    ConfigGVR ConfigGVR `json:"configGVR,omitempty"`
+    
+    // defaultConfig represents the namespace and name of the default add-on configuration.
+    // In scenario where all add-ons have a same configuration.
+    // +optional
+    DefaultConfig ConfigReferent `json:"defaultConfig,omitempty"`
+}
+
+// ConfigGVR represents the GroupVersionResource of the add-on configuration
+type ConfigGVR struct {
+    // group of the add-on configuration.
+    Group string `json:"group"`
+
+    // version of the add-on configuration.
+    Version string `json:"version"`
+
+    // resource of the add-on configuration.
+    Resource string `json:"resource"`
+}
+
+// ConfigReferent represents the namespace and name for an add-on configuration.
+type ConfigReferent struct {
+    // namespace of the add-on configuration.
+    // If this field is not set, the configuration is in the cluster scope.
+    // +optional
+    Namespace string `json:"namespace,omitempty"`
+
+    // name of the add-on configuration.
+    Name string `json:"name"`
+}
+```
+
+When the `ConfigCoordinates` is defined in the `ClusterManagementAddOnSpec`, the addon-framework will sync the
+information of `ConfigCoordinates` to the `ManagedClusterAddOn` status.
+
+In the `ManagedClusterAddOn` spec, we will add a `config` field, it is used to specify the add-on configuration
+namespace and name, and introduce a `ConfigReference` instead of the `ConfigCoordinates` in the
+`ManagedClusterAddOn` status
+
+```golang
+// ManagedClusterAddOnSpec defines the install configuration of an add-on agent on managed cluster.
+type ManagedClusterAddOnSpec struct {
+    // installNamespace is the namespace on the managed cluster to install the add-on agent.
+    // If it is not set, open-cluster-management-agent-addon namespace is used to install the add-on agent.
+    // +optional
+    // +kubebuilder:validation:MaxLength=63
+    // +kubebuilder:validation:Pattern=^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
+    InstallNamespace string `json:"installNamespace,omitempty"`
+
+    // config represents the configuration namespace and name for the current add-on.
+    // In scenario where the current add-on has its own configuration.
+    // +optional
+    Config ConfigReferent `json:"config,omitempty"`
+}
+
+// ManagedClusterAddOnStatus provides information about the status of the operator.
+// +k8s:deepcopy-gen=true
+type ManagedClusterAddOnStatus struct {
+    // Deprecated: Use configReference instead
+    // addOnConfiguration is a reference to configuration information for the add-on.
+    // This resource is use to locate the configuration resource for the add-on.
+    // +optional
+    AddOnConfiguration ConfigCoordinates `json:"addOnConfiguration",omitempty`
+
+    // configReference is a reference to the add-on configuration.
+    // +optional
+    ConfigReference ConfigReference `json:"configReference,omitempty"`
```

Comment:
only the status will have the LastObservedGeneration, right?

Reply:
yes, it's right